### ISSUE TITLE
Java null pointer exception 

### ISSUE THREAD

**Issue by ganeshnikumbh** [OpenLineage/OpenLineage#536]
Hi,

I have installed openlineage on databricks and keep getting below error . Can you please let me know what could be the issue here?  

22/02/12 18:50:26 ERROR AsyncEventQueue: Listener OpenLineageSparkListener threw an exception
java.lang.NullPointerException
	at io.openlineage.spark.agent.lifecycle.ContextFactory.createSparkSQLExecutionContext(ContextFactory.java:32)
	at io.openlineage.spark.agent.OpenLineageSparkListener.lambda$getSparkSQLExecutionContext$4(OpenLineageSparkListener.java:172)
	at java.util.HashMap.computeIfAbsent(HashMap.java:1127)
	at java.util.Collections$SynchronizedMap.computeIfAbsent(Collections.java:2674)
	at io.openlineage.spark.agent.OpenLineageSparkListener.getSparkSQLExecutionContext(OpenLineageSparkListener.java:171)
	at io.openlineage.spark.agent.OpenLineageSparkListener.sparkSQLExecStart(OpenLineageSparkListener.java:125)
	at io.openlineage.spark.agent.OpenLineageSparkListener.onOtherEvent(OpenLineageSparkListener.java:117)
	at org.apache.spark.scheduler.SparkListenerBus.doPostEvent(SparkListenerBus.scala:102)
	at org.apache.spark.scheduler.SparkListenerBus.doPostEvent$(SparkListenerBus.scala:28)
	at org.apache.spark.scheduler.AsyncEventQueue.doPostEvent(AsyncEventQueue.scala:37)
	at org.apache.spark.scheduler.AsyncEventQueue.doPostEvent(AsyncEventQueue.scala:37)
	at org.apache.spark.util.ListenerBus.postToAll(ListenerBus.scala:119)
	at org.apache.spark.util.ListenerBus.postToAll$(ListenerBus.scala:103)
	at org.apache.spark.scheduler.AsyncEventQueue.super$postToAll(AsyncEventQueue.scala:105)
	at org.apache.spark.scheduler.AsyncEventQueue.$anonfun$dispatch$1(AsyncEventQueue.scala:105)
	at scala.runtime.java8.JFunction0$mcJ$sp.apply(JFunction0$mcJ$sp.java:23)
	at scala.util.DynamicVariable.withValue(DynamicVariable.scala:62)
	at org.apache.spark.scheduler.AsyncEventQueue.org$apache$spark$scheduler$AsyncEventQueue$$dispatch(AsyncEventQueue.scala:100)
	at org.apache.spark.scheduler.AsyncEventQueue$$anon$2.$anonfun$run$1(AsyncEventQueue.scala:96)
	at org.apache.spark.util.Utils$.tryOrStopSparkContext(Utils.scala:1562)
	at org.apache.spark.scheduler.AsyncEventQueue$$anon$2.run(AsyncEventQueue.scala:96)

**Comment by ganeshnikumbh** [comment]
I am attaching the full error here. The issue is, I am getting the lineage events but the outputs facet is missing the data. I am only seeing inputs.

[](url
[openLineageError.txt](https://github.com/OpenLineage/OpenLineage/files/8058937/openLineageError.txt)
)

**Comment by ganeshnikumbh** [comment]
@collado-mike @wslulciuc pls help!

**Comment by collado-mike** [comment]
The actual cause for the lack of lineage is this error in your logs:

```
22/02/13 19:17:03 ERROR AsyncEventQueue: Listener OpenLineageSparkListener threw an exception
Failure to initialize configurationInvalid configuration value detected for fs.azure.account.key
```

Unfortunately, we can't help debug this without any information from you about your job, your setup, your environment, etc. For instance, you didn't mention that you're running a Databricks cluster on Azure and trying to (read? write?) access a dataset on ABFS.

In order to help debug this, we need as much information as you can share about your job. A bare bones reproduction case is the most helpful. This includes the Spark configuration you use for setting up your Azure credentials, the set up for your ABFS mount points, etc. If you can share this repro case, we can try to help debug.

**Comment by ganeshnikumbh** [comment]
Hi @collado-mike , thanks for your feedback. The issue I am facing is strange. Irrespective of these errors, I do get the lineage events from Databricks jobs, I am seeing below things

1. If I try to save the data to ADLS storage using "delta" format, I get the lineage but the output facet is empty. Here is the command I am using 
"ddf.write.format("delta").mode("overwrite").partitionBy('partition_name').save(adls_path)"

2. If I try to save the data to ADLS storage using "parquet" format, I do see the data in output facet with proper dataset name. Here is the command I am using for it 
"ddf.write.format("parquet").mode("overwrite").save("abfss://pint-rdp-d-01@rdpd01dls2.dfs.core.windows.net/test_data/creditfacility")" 

3. If I try to save the data using "delta" format to dbfs storage, then I see the output facet with data.

So in summary, the combination of ADLS storage with saving data in delta format is not working.

Any idea on this?

**Comment by ganeshnikumbh** [comment]
closing this as this is the same issue #458 and #459 and will be closed by #491 . Thanks @collado-mike
